### ISSUE TITLE
Not getting prediction correctly using the model trained on the custom dataset (similar format as CORD-V2 dataset) 

### ISSUE THREAD

**Issue by SiriusPoint** [clovaai/donut#297]
I have trained the Donut model using custom dataset which is on the same line as CORD-v2 dataset. The image is having multiple values in one line and we have around 23 to 24 lines in each document. I have used the base model as "naver-clova-ix/donut-base". 
I am using 149 documents for the training and following is the breakup of the datasets
training = 119 images
validation = 22 images
testing = 8 images

I have crated 3 meradata.jsonl file i.e. for train, validation and test. Below is the sample value from the metadat.jsonl file from the training database

`{"file_name": "IOB_Bank_31_image_0.jpg", "ground_truth": "{\"gt_parse\": {\"bank_stmt_entries\": [{\"TXN_DATE\": \"02-11-2023\", \"TXN_DESC\": \"SB Int: 10-2023:0\", \"CHEQUE_REF_NO\": null, \"WITHDRAWAL_AMT\": null, \"DEPOSIT_AMT\": \"93.00\", \"BALANCE_AMT\": \"10901.92\"}, {\"TXN_DATE\": \"09-12-2023\", \"TXN_DESC\": \"CHRGS- SMS ALERT\", \"CHEQUE_REF_NO\": null, \"WITHDRAWAL_AMT\": \"1.06\", \"DEPOSIT_AMT\": null, \"BALANCE_AMT\": \"10900.86\"}, {\"TXN_DATE\": \"02-02-2024\", \"TXN_DESC\": \"Debit Card AMC-2\", \"CHEQUE_REF_NO\": null, \"WITHDRAWAL_AMT\": \"295.00\", \"DEPOSIT_AMT\": null, \"BALANCE_AMT\": \"10605.86\"}, {\"TXN_DATE\": \"02-02-2024\", \"TXN_DESC\": \"SB Int: 01-2024: 0\", \"CHEQUE_REF_NO\": null, \"WITHDRAWAL_AMT\": null, \"DEPOSIT_AMT\": \"75,00\", \"BALANCE_AMT\": \"10680.86\"}]}}"}`

I trained the model for 30 epochs and following are the values for loss and val_edit_distance

**loss = 0.03544
val_edit_distance = 0.3443**

Following is the config parameters used for the training 

- "max_epochs":30,
- "val_check_interval":0.2, # how many times we want to validate during an epoch
- "check_val_every_n_epoch":1,
- "gradient_clip_val":1.0,
- "num_training_samples_per_epoch": 119,
- "lr":3e-5,
- "train_batch_sizes": [2],
- "val_batch_sizes": [1],
- "num_nodes": 1,
- "warmup_steps": 180, # 800/8*30/10, 10%
- "result_path": "/content/drive/MyDrive/universal-bank-statement-reader/processed-dataset/result",
- "verbose": True,

When I am trying to find the prediction using the test dataset, I am getting following output because I had put the print statement at specific location

##### score ==>: 0

I had referred following URL as reference
https://github.com/NielsRogge/Transformers-Tutorials/blob/master/Donut/CORD/Fine_tune_Donut_on_a_custom_dataset_(CORD)_with_PyTorch_Lightning.ipynb

Please help me out in identifying and revolve the issue and let me know if you need more information

Thank you in advance

**Comment by CarlosSerrano88** [comment]
@SiriusPoint any updates? I have the same problem

**Comment by SiriusPoint** [comment]
@CarlosSerrano88, Not yet. I am trying but not getting appropriate results.

**Comment by banditgoose** [comment]
The transformers implementation of Donut seems to have broken saving and loading at some point. Try transformers==4.26.1 and see if that works.

**Comment by dreamlychina** [comment]
> any updates? I have the same problem

+1

**Comment by CarlosSerrano88** [comment]
with transformers==4.25.1 working perfect!

**Comment by dgarlor** [comment]
I have the same problem with the last version of transformers. Going back to 4.40.1, and the saved model works again

**Comment by achardmaple** [comment]
> 与 transformers==4.25.1 完美协作！

Did you solve the similar question when training on cord-v2 by using transformers 4.25.1?
